src/alda/sound.clj

```diff
@@ -278,6 +288,61 @@
                                (/ (score-length events) 1000.0)
                                1) end!)))
 
+(defn schedule-wait!
+  "Simple function which waits for a sequencer to finish."
+  [score sequencer wait]
+  (let [{:keys [instruments audio-context]} score
+        engine (:synthesis-engine @audio-context)
+        begin  (.getCurrentTime ^SynthesisEngine engine)
+        end! #(deliver wait :done)]
+    (schedule-event! engine (+ begin
+                               (/ (.getMicrosecondLength sequencer) 1000.0 1000.0)
+                               1) end!)))
```

Comment:
I think it would be nice to figure out how to fully replace JSyn with a Sequencer-based implementation. This PR gets us most of the way there, except for this part where we're still using a JSyn SynthesisEngine to schedule the delivery of the `:done` value to the `wait` promise, giving calling code a way to block until the score is done playing.

I'm sure there is some other way we could do this... If `Sequencer.start()` is synchronous (i.e. it blocks until playback is complete), then it might even be as simple as calling `.start()` and then delivering the `:done` value.

Reply:
Unfortunately, I don't think these sequencers block when they play (from playing around with them), it would be very convenient if they did though. They do have a `isRunning` function though.

I'm going to look around and see if there's a better way we can do this, I'd like to avoid this method too.

Reply:
I googled this a bit, and it seems like the proper way to do this might be to add a "meta-event listener" that reacts when the message type is 0x2f ("end of track") 

I found this in a StackOverflow answer (https://stackoverflow.com/a/5972712/2338327), although I haven't found anything in the official Java docs about these meta-messages. I'm not sure if the 0x2f message is sent by default, but if it's not, it seems like we should at least be able to add it to the track ourselves.